### ISSUE TITLE
Change package names to org.apache.parquet.*

### ISSUE THREAD

**Issue by asfimport** [apache/parquet-java#1430]
Parquet's package names are still `parquet.\*`. Since Parquet is now in the Apache Incubator, the namespaces should be updated to be `org.apache.parquet.\*`.

**Reporter**: [David Chen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=davidzchen)
**Assignee**: [Ryan Blue](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rdblue) / @rdblue
#### Original Issue Attachments:
- [refactor.sh](https://issues.apache.org/jira/secure/attachment/12658315/refactor.sh)

<sub>**Note**: *This issue was originally created as [PARQUET-23](https://issues.apache.org/jira/browse/PARQUET-23). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Dmitriy V. Ryaboy](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14069109) / @dvryaboy:
Is this an actual apache requirement, or simply a convention?

I am not sure how this will help our users, but it will certainly cause migration pain.

**Comment by asfimport** [comment]
[Julien Le Dem](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14070469) / @julienledem:
as we're going to change the maven coordinates we need to change the package name as well.
Otherwise you can have conflicts where 2 artifacts that look distinct in maven pull the same classes.

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14077215) / @rdblue:
I'm attaching my refactor script, which does most of the work and has been verified up through parquet-pig. The parquet-hadoop module needs a lot of changes outside of what is done in the script because there are a lot of code references that use full package names when there are collisions with parquet-format classes.

This is handled correctly (and probably more reliably) by IntelliJ refactoring, but that requires you to move the package in each module separately, for both test and main. This also causes unused import statements in tests, but that is less of a concern than getting the classes wrong in parquet-hadoop. Either way, the refactor is going to require manual steps and will take a couple of hours.

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14221529) / @rdblue:
Opened PR for parquet-format: [#18](https://github.com/apache/incubator-parquet-format/pull/18)

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14226663) / @rdblue:
I think PR #18 is ready for review. I've also posted a [parquet-mr branch](https://github.com/rdblue/incubator-parquet-mr/tree/PARQUET-23-test-org-apache-parquet-format) that can build using a snapshot of #18. The build and tests pass and I can use parquet-tools for existing parquet files so I don't think there is an incompatibility from the parquet-format changes.

The only problem I ran into and still have to track down is a bug in parquet-tools dump:
```Java
[cloudera@quickstart ~]$ hadoop jar parquet-tools-1.6.0rc3-SNAPSHOT.jar dump --debug test.parquet
java.lang.NoSuchMethodError: parquet.hadoop.metadata.ColumnChunkMetaData.getPath()Lparquet/common/schema/ColumnPath;
        at parquet.tools.util.MetadataUtils.showDetails(MetadataUtils.java:92)
        at parquet.tools.command.DumpCommand.dump(DumpCommand.java:173)
        at parquet.tools.command.DumpCommand.execute(DumpCommand.java:134)
        at parquet.tools.Main.main(Main.java:219)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.hadoop.util.RunJar.main(RunJar.java:212)
parquet.hadoop.metadata.ColumnChunkMetaData.getPath()Lparquet/common/schema/ColumnPath;

2014-11-26 10:59:33 PST
```

I don't think this is from the parquet-format changes, but I want to track it down before we merge.

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14499126) / @rdblue:
Submitted [PR #179](https://github.com/apache/incubator-parquet-mr/pull/179) with the rename changes.

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-23?#comment-14518136) / @rdblue:
Merged #182
